asn1/InterledgerTypes.asn

```diff
@@ -0,0 +1,165 @@
+InterledgerTypes
+DEFINITIONS
+AUTOMATIC TAGS ::=
+BEGIN
+
+IMPORTS
+    Int16,
+    UInt8,
+    UInt64,
+    VarUInt,
+    Float64
+    FROM GenericTypes
+;
+
+-- Readable names for special characters that may appear in ILP addresses
+hyphen IA5String ::= "-"
+period IA5String ::= "."
+underscore IA5String ::= "_"
+tilde IA5String ::= "~"
+
+-- A standard interledger address
+Address ::= IA5String
+    (FROM
+        ( hyphen
+        | period
```

Comment:
I don't think that's possible in ASN.1, but am happy to be proven wrong by someone with better ASN.1-fu.